libs/langchain/langchain/indexes/_sql_record_manager.py

```diff
@@ -192,17 +202,37 @@ def update(
         ]
 
         with self._make_session() as session:
-            # Note: uses SQLite insert to make on_conflict_do_update work.
-            # This code needs to be generalized a bit to work with more dialects.
-            insert_stmt = insert(UpsertionRecord).values(records_to_upsert)
-            stmt = insert_stmt.on_conflict_do_update(  # type: ignore[attr-defined]
-                [UpsertionRecord.key, UpsertionRecord.namespace],
-                set_=dict(
-                    # attr-defined type ignore
-                    updated_at=insert_stmt.excluded.updated_at,  # type: ignore
-                    group_id=insert_stmt.excluded.group_id,  # type: ignore
-                ),
-            )
+            if self.dialect == "sqlite":
+                from sqlalchemy.dialects.sqlite import insert as sqlite_insert
+
+                # Note: uses SQLite insert to make on_conflict_do_update work.
+                # This code needs to be generalized a bit to work with more dialects.
+                insert_stmt = sqlite_insert(UpsertionRecord).values(records_to_upsert)
+                stmt = insert_stmt.on_conflict_do_update(  # type: ignore[attr-defined]
+                    [UpsertionRecord.key, UpsertionRecord.namespace],
+                    set_=dict(
+                        # attr-defined type ignore
+                        updated_at=insert_stmt.excluded.updated_at,  # type: ignore
+                        group_id=insert_stmt.excluded.group_id,  # type: ignore
+                    ),
+                )
+            elif self.dialect == "postgresql":
+                from sqlalchemy.dialects.postgresql import insert as pg_insert
+
+                # Note: uses SQLite insert to make on_conflict_do_update work.
+                # This code needs to be generalized a bit to work with more dialects.
+                insert_stmt = pg_insert(UpsertionRecord).values(records_to_upsert)
+                stmt = insert_stmt.on_conflict_do_update(  # type: ignore[attr-defined]
+                    "uix_key_namespace",  # Name of constraint
```

Comment:
Requires using constraint name -- does not support list of columns for some reason. Tests pass with both postgresql and sqlite locally. I'll add postgres tests in a bit